flocker/node/_docker.py

```diff
@@ -564,11 +564,15 @@ def _list():
                 image_data = self._client.inspect_image(image)
                 unit_environment = []
                 container_environment = data[u"Config"][u"Env"]
-                image_environment = image_data[u"Config"]["Env"]
-                for environment in container_environment:
-                    if environment not in image_environment:
-                        env_key, env_value = environment.split('=', 1)
-                        unit_environment.append((env_key, env_value))
+                if image_data[u"Config"]["Env"] is not None:
+                    image_environment = image_data[u"Config"]["Env"]
+                else:
+                    image_environment = []
```

Comment:
:question: 

All other things equal, it's easier to read code without `not` than with it.  Cases like this, you can easily avoid using `not` by flipping the order of the blocks.